test/integration/profiler_test.go

```diff
@@ -504,4 +525,40 @@ func TestCPUProfilerWorks(t *testing.T) {
 			requireAnyStackContains(t, jitStacks, []string{"jit_top", "jit_middle"})
 		}
 	})
+
+	t.Run("unwinder metrics work", func(t *testing.T) {
+		addr := "localhost:7071"
+
+		// Spawn the HTTP server with the /metrics Prometheus handler.
+		mux := http.NewServeMux()
+		mux.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+
+		srv := &http.Server{
+			Addr:         addr,
+			Handler:      mux,
+			ReadTimeout:  1 * time.Second,
+			WriteTimeout: 1 * time.Second,
+		}
+		go srv.ListenAndServe()
+		t.Cleanup(func() {
+			srv.Shutdown(context.Background())
+		})
+
+		resp, err := http.Get(fmt.Sprintf("http://%s/metrics", addr)) //nolint: noctx
+		require.Nil(t, err)
+		t.Cleanup(func() {
+			resp.Body.Close()
+		})
+
+		body, err := io.ReadAll(resp.Body)
+		require.Nil(t, err)
+
+		metrics := parsePrometheusMetricsEndpoint(string(body))
+		require.Greater(t, len(metrics), 0)
+
+		// TODO: fix this assertion, all the BPF map metrics are zero but I am not sure why.
```

Comment:
Will create an issue to track this. As this works while running the Agent, I think this might be caused by a wrong setup/teardown of the code during testing. Spent some time debugging this but I haven't made much progress unfortunately